tracing-subscriber/src/fmt/format/mod.rs

```diff
@@ -1526,23 +1536,60 @@ pub(super) mod test {
             .with_timer(MockTime);
         let expected = "fake time tracing_subscriber::fmt::format::test: hello\n";
 
-        run_test(subscriber, make_writer, expected);
+        assert_info_hello(subscriber, make_writer, expected);
     }
 
-    #[cfg(feature = "ansi")]
-    fn test_ansi(is_ansi: bool, expected: &str) {
+    #[test]
+    fn with_thread_ids() {
         let make_writer = MockMakeWriter::default();
         let subscriber = crate::fmt::Collector::builder()
             .with_writer(make_writer.clone())
-            .with_ansi(is_ansi)
+            .with_thread_ids(true)
+            .with_ansi(false)
+            .with_timer(MockTime);
+        let expected = "fake time  INFO ThreadId() tracing_subscriber::fmt::format::test: hello\n";
+
+        assert_info_hello_ignore_numeric(subscriber, make_writer, expected);
+    }
+
+    #[test]
+    fn pretty_default() {
+        let make_writer = MockMakeWriter::default();
+        let subscriber = crate::fmt::Collector::builder()
+            .pretty()
+            .with_writer(make_writer.clone())
+            .with_ansi(false)
             .with_timer(MockTime);
-        run_test(subscriber, make_writer, expected)
+        let expected = r#"  fake time  INFO tracing_subscriber::fmt::format::test: hello
+    at tracing-subscriber/src/fmt/format/mod.rs:
+
+"#
+        .replace('/', &std::path::MAIN_SEPARATOR.to_string());
+
+        assert_info_hello_ignore_numeric(subscriber, make_writer, &expected)
     }
 
-    fn run_test(subscriber: impl Into<Dispatch>, buf: MockMakeWriter, expected: &str) {
+    fn assert_info_hello(subscriber: impl Into<Dispatch>, buf: MockMakeWriter, expected: &str) {
         let _default = set_default(&subscriber.into());
         tracing::info!("hello");
-        assert_eq!(expected, buf.get_string())
+        let result = buf.get_string();
+
+        assert_eq!(expected, result)
+    }
+
+    // When numeric characters are used they often form a non-deterministic value as they usually represent things like a thread id or line number.
+    // This assert method should be used when non-deterministic numeric characters are present.
+    fn assert_info_hello_ignore_numeric(
+        subscriber: impl Into<Dispatch>,
+        buf: MockMakeWriter,
+        expected: &str,
+    ) {
+        let _default = set_default(&subscriber.into());
+        tracing::info!("hello");
+
+        let result = buf.get_string().replace(char::is_numeric, "");
```

Comment:
The reason I originally suggested using a regular expression here is that this approach does have the issue that the tests will _still pass_ if the line numbers (and other numeric characters) are totally missing, which is not really ideal. 

I don't actually think it's important to avoid adding the `regex` crate as a dev-dependency, because...`tracing-subscriber` already _has_ a dev-dependency on `regex`: https://github.com/tokio-rs/tracing/blob/6dd9d236fa17aa4b72eddc36a89d230c66140466/tracing-subscriber/Cargo.toml#L71 :)

So, I think changing this to use a regular expression is probably the best solution.

Reply:
ahaha I completely missed it was already a dep.
I changed it to use a regex.